sdk/shared/src/main/scala/org/ergoplatform/sdk/utils/Zero.scala

```diff
@@ -0,0 +1,144 @@
+
+  implicit val BooleanIsZero: Zero[Boolean] = CZero(false)
+  implicit val ByteIsZero: Zero[Byte] = CZero(0.toByte)
+  implicit val ShortIsZero: Zero[Short] = CZero(0.toShort)
+  implicit val IntIsZero: Zero[Int] = CZero(0)
+  implicit val LongIsZero: Zero[Long] = CZero(0L)
+  implicit val BigIntIsZero: Zero[BigInt] = CZero(CBigInt(BigInteger.ZERO))
+  implicit val GroupElementIsZero: Zero[GroupElement] = CZero(CGroupElement(CryptoConstants.dlogGroup.identity))
+  implicit val AvlTreeIsZero: Zero[AvlTree] = CZero({
+    val avlProver = new BatchAVLProver[Digest32, Blake2b256.type](keyLength = 32, None)
+    val digest = avlProver.digest
+    val treeData = new AvlTreeData(digest, AvlTreeFlags.AllOperationsAllowed, 32, None)
+    CAvlTree(treeData)
+  })
+  implicit val sigmaPropIsZero: Zero[SigmaProp] = CZero(CSigmaProp(TrivialProp.FalseProp))
+  implicit val AnyIsZero: Zero[Any] = CZero(0)
+  implicit val UnitIsZero: Zero[Unit] = CZero(())
+  implicit val SigmaDslBuilderIsZero: Zero[SigmaDslBuilder] = CZero(CostingSigmaDslBuilder)
+  implicit val PreHeaderIsZero: Zero[PreHeader] = CZero({
+    CPreHeader(
+      ByteIsZero.zero,
+      Colls.emptyColl[Byte],
+      LongIsZero.zero,
+      LongIsZero.zero,
+      IntIsZero.zero,
+      GroupElementIsZero.zero,
+      Colls.emptyColl[Byte]
+    )
+  })
+  implicit val BoxIsZero: Zero[Box] = CZero({
+    new ErgoBox(
+      LongIsZero.zero,
+      new ErgoTree(
+        ByteIsZero.zero,
+        IndexedSeq.empty,
+        Right(sigmaPropIsZero.zero)
+      ),
+      Colls.emptyColl,
+      Map.empty,
+      ModifierId @@ ("synthetic_transaction_id"),
+      ShortIsZero.zero,
+      IntIsZero.zero
+    )
+  })
+  implicit val ContextIsZero: Zero[Context] = CZero({
+    CostingDataContext(
+      _dataInputs = Colls.emptyColl,
+      headers = Colls.emptyColl,
+      preHeader = PreHeaderIsZero.zero,
+      inputs = Colls.emptyColl,
+      outputs = Colls.emptyColl,
+      height = IntIsZero.zero,
+      selfBox = CostingBox(BoxIsZero.zero),
+      selfIndex = IntIsZero.zero,
+      lastBlockUtxoRootHash = AvlTreeIsZero.zero,
+      _minerPubKey = Colls.emptyColl[Byte],
+      vars = Colls.emptyColl,
+      activatedScriptVersion = ByteIsZero.zero,
+      currentErgoTreeVersion = ByteIsZero.zero
+    )
+  })
+  implicit val HeaderIsZero: Zero[Header] = CZero({
+    CHeader(
+      Colls.emptyColl[Byte],
+      ByteIsZero.zero,
+      Colls.emptyColl[Byte],
+      Colls.emptyColl[Byte],
+      CAvlTree(AvlTreeIsZero.zero),
+      Colls.emptyColl[Byte],
+      timestamp = LongIsZero.zero,
+      nBits = LongIsZero.zero,
+      height = IntIsZero.zero,
+      extensionRoot = Colls.emptyColl[Byte],
+      minerPk = GroupElementIsZero.zero,
+      powOnetimePk = GroupElementIsZero.zero,
+      powNonce = Colls.emptyColl[Byte],
+      powDistance = BigIntIsZero.zero,
+      votes = Colls.emptyColl[Byte]
+    )
+  })
+
+  def typeToZero[T](t: RType[T]): Zero[T] = (t match {
```

Comment:
After looking more closely, I am wondering if I should update `typeToZero` method to take `SType` instead of `RType`.

This way I'll be able to avoid converting from `SType` to `RType` before calling this function.

` DeserializationSigmaBuilder.mkConstant(Zero.zeroOf(Zero.typeToZero(Evaluation.stypeToRType(t))), t)
`

Reply:
Yes, make sense, I wonder what can be the signature?